qiskit/circuit/library/standard_gates/x.py

```diff
@@ -1011,14 +764,96 @@ def control(
+        ctrl_state = _ctrl_state_to_int(ctrl_state, num_ctrl_qubits)
+        new_ctrl_state = (self.ctrl_state << num_ctrl_qubits) | ctrl_state
+        gate = MCXGate(
+            num_ctrl_qubits=num_ctrl_qubits + self.num_ctrl_qubits,
+            label=label,
+            ctrl_state=new_ctrl_state,
+            synthesis=self.synthesis,
+            _name=self.name,
+        )
+        gate.base_gate.label = self.label
+        return gate
+
+    @staticmethod
+    def get_num_ancilla_qubits(num_ctrl_qubits: int, mode: str = "noancilla") -> int:
+        """Get the number of required ancilla qubits without instantiating the class.
+
+        This static method might be necessary to check the number of ancillas before
+        creating the gate, or to use the number of ancillas in the initialization.
+        """
+        # pylint: disable=cyclic-import
+        from qiskit.synthesis.mcx_synthesis import mcx_mode_to_num_ancilla_qubits
+
+        return mcx_mode_to_num_ancilla_qubits(num_ctrl_qubits, mode)
+
+    def __array__(self, dtype=None):
+        """Return a numpy.array for the MCX gate."""
+        mat = _compute_control_matrix(
+            self.base_gate.to_matrix(), self.num_ctrl_qubits, ctrl_state=self.ctrl_state
+        )
+        if dtype:
+            return numpy.asarray(mat, dtype=dtype)
+        return mat
+
+
+# The following explicit classes remain for backward compatibility.
+# The __new__ method is modified to return the MCX gate with the matching synthesis algorithm
+# (or, as previously, an explicit few-qubit gate).
+# These classes still inherit from MCXGate and in particular from ControlledGate.
+
+
+class C4XGate(ControlledGate):
+    """The 3-qubit controlled X gate.
+    The default MCXSynthesisGrayCode algorithm implements the optimized method for 4 control qubits.
+    """
+
+    # pylint: disable=signature-differs
+    def __new__(
+        cls,
+        label: Optional[str] = None,
+        ctrl_state: Optional[Union[str, int]] = None,
+    ):
+        """Create a new MCX instance"""
+        # pylint: disable=cyclic-import
+        from qiskit.synthesis.mcx_synthesis import MCXSynthesisGrayCode
+
+        gate = MCXGate(
+            num_ctrl_qubits=4,
+            label=label,
+            ctrl_state=ctrl_state,
+            synthesis=MCXSynthesisGrayCode(),
+            _name="mcx",
+        )
+        return gate
+
+
+class C3XGate(ControlledGate):
+    """The 3-qubit controlled X gate.
+    The default MCXSynthesisGrayCode algorithm implements the optimized method for 3 control qubits.
+    """
+
+    # pylint: disable=signature-differs
+    def __new__(
+        cls,
+        angle: Optional[ParameterValueType] = None,
+        label: Optional[str] = None,
+        ctrl_state: Optional[Union[str, int]] = None,
+    ):
+        if angle is not None:
+            return C3SXGate(label, ctrl_state, angle=angle)
+
+        from qiskit.synthesis.mcx_synthesis import MCXSynthesisGrayCode
```

Comment:
I think this should directly call the 3-qubit synthesis for the C3X, otherwise it seems a bit backwards: the C3X calls the general Gray code which, as a special case, happens to have the 3-qubit synthesis we want here 🙂 (Same for the C4X)